### PR TITLE
10/ejb

### PR DESCRIPTION
**Pull request by selhanaf**

### CODE REVIEW

File: src/main/java/com/car/app/controller/CarDao.java [hunk 1]
```diff
@@ -62,20 +63,12 @@ public Car findCarById(String id) {
 	 * @return
 	 * @throws Exception 
 	 */
-	public Car createCar(Car car) throws Exception {
+	@Transactional(REQUIRED)
```
Comment (darkjeffbr) [inline src/main/java/com/car/app/controller/CarDao.java:66]: The default value for transaction is REQUIRED, you don't need to remove, just to let you know.

File: src/main/java/com/car/app/controller/CarDao.java [hunk 2]
```diff
@@ -108,6 +95,7 @@ public Car updateCar(Car car) throws Exception {
 	 * @param carId
 	 * @return
 	 */
+	@Transactional(REQUIRED)
 	public boolean deleteCar(String carId) {
 		log.info("ENTER : deleteCar");
 		log.info("remove car with id ={} ", carId);
```
Comment (darkjeffbr) [inline src/main/java/com/car/app/controller/CarDao.java:101]: Dealing with transaction here ? I think you missed to update this method 😄

File: src/main/java/com/car/app/controller/CarService.java [hunk 3]
```diff
@@ -2,14 +2,22 @@
...
 import org.slf4j.LoggerFactory;
 
 import com.car.app.model.Car;
 
+@Stateless
 public class CarService implements ICarService {
 	private static Logger log = LoggerFactory.getLogger(CarService.class);
-	private CarDao carDao = new CarDao();
+	
+	@EJB
```
Comment (darkjeffbr) [inline src/main/java/com/car/app/controller/CarService.java]: Usually we use @Inject to inject a bean, using EJB is to cases where we have a remote bean or a circular dependency or cases where @Inject is not capable of handling the injection. It is not wrong to use EJB, just unusual.

File: src/main/java/com/car/app/controller/ICarDao.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.util.List;
+
+import javax.ejb.Remote;
+import javax.transaction.Transactional;
+
+import com.car.app.model.Car;
+
+
+@Remote
```
Comment (darkjeffbr) [inline src/main/java/com/car/app/controller/ICarDao.java]: Why annotating this interface as @Remote ? Not needed.

File: src/main/java/com/car/app/boundary/RestApplication.java [hunk 5]
```diff
@@ -28,7 +28,7 @@ public RestApplication() {
 	        OpenAPI oas = new OpenAPI();
 	        oas.setServers(null);
 	        Server server = new Server();
-	        server.setUrl("http://localhost:8080/carapp");
+	        server.setUrl("/carapp");
```
Comment (darkjeffbr) [inline src/main/java/com/car/app/boundary/RestApplication.java:31]: Nice that you have removed the hard coded server address and port. Did you check the Application class ?

**Review by darkjeffbr — CHANGES_REQUESTED** [review]

File: src/main/java/com/car/app/controller/CarService.java (hunk 3 above)
Comment (selhanaf) [reply src/main/java/com/car/app/controller/CarService.java]: Ok, let's use the usual :D

**Review by darkjeffbr — APPROVED** [review]

**Pull request merged**